app/controllers/game_controller.rb

```diff
@@ -1,35 +1,38 @@
 class GameController < ApplicationController
   skip_before_action :verify_authenticity_token
 
-  # def start
-  def initilize
-    super
-    @lives ||= 9
-    @words ||= %w[programming fun internship ruby rails coding design]
-
-    @word_to_guess ||= @words.sample
-    @hidden_word = @word_to_guess.chars.map { '_' }
+  def start
+    words = %w[programming fun internship ruby rails coding design]
+    game = GameSession.new(word_to_guess: words.sample, guesses: '')
+    game.save
+    render('start')
   end
 
   def word_guess
-    @user_guess = params[:guess] # .chomp.downcase
+    user_guess = params[:guess] # .chomp.downcase
+    game = GameSession.last
+    guesses = game.guesses
+    game.update(guesses: guesses + user_guess)
+    @hidden_word = game.get_hidden_word
 
-    if @word_to_guess.include? @user_guess
-      @hidden_word = @hidden_word.filter_map.with_index do |letter, index|
-        @user_guess == @word_to_guess[index] ? @user_guess : letter
-      end
-    else
-      @lives -= 1
-    end
-    if @hidden_word.join('') == @word_to_guess
-      @message = 'You won the game!'
-      render('result')
+    @lives = game.lives
+    if game.win?
+      redirect_to(:action => 'result', :message => "Yay! you won the game :)", :word => game.word_to_guess)
```

Comment:
Far more common to use the new hash syntax these days:

```suggestion
      redirect_to(action: 'result', message: 'Yay! you won the game :)', word: game.word_to_guess)
```

EDIT: I think rubocop will ping you on this actually